ext/plugins/lint-nested-interactive.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+/*
+ Disallows nested of interactive elements
+
+ ```
+ {{! good }}
+ <button>Click here</button> <a href="/">and a link</a>
+
+ {{! bad}}
+ <button>Click here <a href="/">and a link</a></button>
+ ```
+
+ The following values are valid configuration:
+
+   * boolean -- `true` for enabled / `false` for disabled
+ */
+
+var calculateLocationDisplay = require('../helpers/calculate-location-display');
+var buildPlugin = require('./base');
+
+module.exports = function(addonContext) {
+  var LogNestedInteractive = buildPlugin(addonContext, 'nested-interactive');
+
+  function isElementNode(node) {
+    return node.type === 'ElementNode';
+  }
+
+  LogNestedInteractive.prototype.detect = function(node) {
+    return node.type === 'Program';
+  };
+
+  LogNestedInteractive.prototype.process = function(node) {
+    node.body.filter(isElementNode).forEach(function(node) {
+      this.findNestedInteractiveElements(node, false);
+    }, this);
+  };
+
+  LogNestedInteractive.prototype.findNestedInteractiveElements = function(node, parentInteractiveTag) {
+    /** @type {string} [=""] */
+    parentInteractiveTag = parentInteractiveTag || '';
+
+    if (Array.isArray(node.children) && node.children.length === 0) {
+      return;
+    }
+
+    /** @type {string} */
+    var currentTag = node.tag;
+
+    if (parentInteractiveTag === 'a' || parentInteractiveTag === 'button') {
+      if (currentTag === 'a' || currentTag === 'button') {
```

Comment:
there is a technical issue with `label` because it's common and okay to put an `<input>` inside `<label>`. i might count `label` in but in case there is `for` tag for the `label`